libs/wallet/src/errors.ts

```diff
@@ -0,0 +1,22 @@
+export class ConnectorError extends Error {
+  code: number;
+
+  constructor(message: string, code: number) {
+    super(message);
+    this.name = 'ConnectorError';
+    this.code = code;
+  }
+}
+
+export const ConnectorErrors = {
+  userRejected: new ConnectorError('user rejected', 0),
+  noConnector: new ConnectorError('no connector', 1),
+  connect: new ConnectorError('failed to connect', 2),
+  disconnect: new ConnectorError('failed to disconnect', 3),
+  chainId: new ConnectorError('incorrect chain', 4),
+  listKeys: new ConnectorError('failed to list keys', 5),
+  isConnected: new ConnectorError('failed to check connection', 6),
+  sendTransaction: new ConnectorError('failed to send transaction', 7),
+  unknown: new ConnectorError('unknown error', 8),
+  noWallet: new ConnectorError('no wallet running', 9),
+};
```

Comment:
Creating the errors here means we always throw the same instance of the error. Might be better to store the message + code and create the errors before throwing. I've never seen this before, not sure what the result is possibly incorrect stack?

Reply:
Addressed this in this commit: https://github.com/vegaprotocol/frontend-monorepo/pull/5815/commits/99e5efbd88ca92f4b10774b46f72cf4a04b5b870